### PR TITLE
Include discount in fees brakedown tooltip

### LINKED ISSUE #7750: Discounted fee details in tooltip do not match the discount shown in pill
### Describe the bug

When onboarding with a discounted fee associated with an incentive, the discount is displayed properly in the pill, but the tooltip details that appear on mousehover do not show the discount.

This is similar to #5255 but this could be an issue that specifically affects discounted rates that are enabled due to an incentive. Flagging this to @nagpai since he's done a lot of work on fees, discounts, and how these pills display that information.

### To Reproduce

1. Onboard a WCPay account with discounted card fees associated with an incentive.
2. Navigate to **Payments > Settings.**
3. Mouseover the `Credit Debit card` pill with the discounted rate: `From x.xx% + $x.xx — see Terms and Conditions`
4. The additional details listed in the tooltip do not include the non-discounted `Base fee`; the normal, non-discounted base fee is listed instead.

### Actual behavior
The details of the discounted rate are not accurately displayed in the fee detail element, 

### Screenshots

My site qualifies for a 10% discount (2.61% + $0.27)...

wcpay 10 discount

...but the base US fee of 2.9% + $0.30 displays in the tooltip:

custom rate discrepancy

### Expected behavior
Custom rates are a part of the `Base fee` when viewing the fee details tooltip.

### PR DESCRIPTION
**Pull request by KarlisJ**
Fixes #7750 

#### Changes proposed in this Pull Request

Applies the discount rate to all fees in the tooltip so that the values match with actual fees. 

#### Testing instructions

1. Add any custom fee to any payment method
2. Go to `payments` -> `settings` 
3. The fees pill should have `From ...` 
4. That values should match the base fee in the tooltip
-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7787` or branch name `fix/7750-include-discount-in-tooltip` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7750-include-discount-in-tooltip) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 9caf645aedc66294080338cf1a932e938a690fce
- Build time: 2023-12-08 12:52:22 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/utils/account-fees.tsx [hunk 1]
```diff
@@ -74,24 +74,27 @@ const getStripeFeeSectionUrl = ( country: string ): string => {
...
-			formatCurrency( fee.fixed_rate, fee.currency )
+			formatFee( fee.percentage_rate * discountRate ),
+			formatCurrency( fee.fixed_rate * discountRate, fee.currency )
 		);
 	} else if ( fee.fixed_rate ) {
 		return sprintf(
 			'%2$s',
-			formatFee( fee.percentage_rate ),
-			formatCurrency( fee.fixed_rate, fee.currency )
+			formatFee( fee.percentage_rate * discountRate ),
```
Comment (nagpai) [inline client/utils/account-fees.tsx]: Do we need `formatFee( fee.percentage_rate * discountRate )` here within the `sprintf` statement, for a condition when it is not available?  Will this be better: 

```suggestion
if ( fee.fixed_rate ) {
		return sprintf(
			'%1$s',
			formatCurrency( fee.fixed_rate * discountRate, fee.currency )
		);
```

File: client/utils/account-fees.tsx [hunk 2]
```diff
@@ -74,24 +74,27 @@ const getStripeFeeSectionUrl = ( country: string ): string => {
...
+			formatCurrency( fee.fixed_rate * discountRate, fee.currency )
 		);
 	} else if ( fee.percentage_rate ) {
 		return sprintf(
 			'%1$f%%',
-			formatFee( fee.percentage_rate ),
-			formatCurrency( fee.fixed_rate, fee.currency )
+			formatFee( fee.percentage_rate * discountRate ),
```
Comment (nagpai) [inline client/utils/account-fees.tsx]: Similar to the comment on line 87-90, Do we need to have `formatCurrency( fee.fixed_rate * discountRate, fee.currency ) within the `sprintf` statement , when we do not expect it? 

```suggestion
else if ( fee.percentage_rate ) {
		return sprintf(
			'%1$f%%',
			formatFee( fee.percentage_rate * discountRate ),
        );
```

File: client/utils/account-fees.tsx [hunk 3]
```diff
@@ -109,19 +112,19 @@ export const formatMethodFeesTooltip = (
 	accountFees: FeeStructure
 ): JSX.Element => {
 	if ( ! accountFees ) return <></>;
-	const currentBaseFee = getCurrentBaseFee( accountFees );
-	// If the current fee doesn't have a fixed or percentage rate, use the base fee's rate. Eg. when there is a promotional discount fee applied. Use this to calculate the total fee too.
-	const currentFeeWithBaseFallBack = currentBaseFee.percentage_rate
-		? currentBaseFee
-		: accountFees.base;
+
+	const discountFeeRate: number =
```
Comment (nagpai) [inline client/utils/account-fees.tsx]: Thanks for fixing this!

**Review by nagpai — CHANGES_REQUESTED** [review]
A couple of suggestions. Else LGTM!

File: client/utils/account-fees.tsx [hunk 4]
```diff
@@ -74,24 +74,27 @@ const getStripeFeeSectionUrl = ( country: string ): string => {
 	);
 };
 
-const getFeeDescriptionString = ( fee: BaseFee ): string => {
+const getFeeDescriptionString = (
+	fee: BaseFee,
+	discountRate: number
```
Comment (dmvrtx) [inline client/utils/account-fees.tsx]: Based on the usage it should be `discountedRate` or `discountAdjustedFeeRate`.

File: client/utils/account-fees.tsx [hunk 5]
```diff
@@ -109,19 +112,19 @@ export const formatMethodFeesTooltip = (
...
-	const currentBaseFee = getCurrentBaseFee( accountFees );
-	// If the current fee doesn't have a fixed or percentage rate, use the base fee's rate. Eg. when there is a promotional discount fee applied. Use this to calculate the total fee too.
-	const currentFeeWithBaseFallBack = currentBaseFee.percentage_rate
-		? currentBaseFee
-		: accountFees.base;
+
+	const discountFeeRate: number =
+		accountFees.discount.length && accountFees.discount[ 0 ].discount
+			? 1 - accountFees.discount[ 0 ].discount
+			: 1;
```
Comment (dmvrtx) [inline client/utils/account-fees.tsx]: As mentioned above, it is `discountedFeeRate` or, even better, `discountAdjustedFeeRate`.

**Review by dmvrtx — COMMENTED** [review]
Code-wise I concur with comments by Nagesh and would like to have a better naming for discount-adjusted fee rate variable. Will test it in details later. I'm curious to check how it will work with custom base fees.

File: client/utils/account-fees.tsx (hunk 4 above)
Comment (KarlisJ) [reply client/utils/account-fees.tsx]: I'm not sure I follow your reasoning here. I definitely don't like using "adjusted" in the naming as it implies that the discount rate is adjusted to something. 

I think this is `discountRate` that comes from discountFee, that's it.

Comment (dmvrtx) [reply client/utils/account-fees.tsx]: My concern is that `discountRate` is used in the fee description already and means how much of the total is the discount, i.e. 20%.

But here it is passed as already calculated/adjusted final rate, i.e. 80%, but with the same name. Thus, I see `discountRate` name as confusing.

Comment (KarlisJ) [reply client/utils/account-fees.tsx]: Renamed in https://github.com/Automattic/woocommerce-payments/pull/7787/commits/a38fc24cb4334e32e5d18112233cf010cf4c8e79 to `discountAdjustedFeeRate`

**Review by nagpai — APPROVED** [review]
LGTM! Thanks.

File: client/utils/account-fees.tsx [hunk 6]
```diff
@@ -74,24 +74,25 @@ const getStripeFeeSectionUrl = ( country: string ): string => {
 	);
 };
 
-const getFeeDescriptionString = ( fee: BaseFee ): string => {
+const getFeeDescriptionString = (
+	fee: BaseFee,
+	discountRate: number
```
Comment (kalessil) [inline client/utils/account-fees.tsx]: I like the solution, but I also find `discountRate` misleading: it's a kinda `discountBasedMultiplier`.

**Review by kalessil — CHANGES_REQUESTED** [review]

File: client/utils/account-fees.tsx (hunk 6 above)
Comment (KarlisJ) [reply client/utils/account-fees.tsx]: After some thoughts, refactored to `discountBasedMultiplier` in https://github.com/Automattic/woocommerce-payments/pull/7787/commits/1083bd4a5ece63704ce39c3a3dbf02d18b5c4faf

**Review by kalessil — APPROVED** [review]
LGTM! Please note some of JS tests are failing (ping me in Slack if support is needed).

Tested cases:
- base fee :heavy_check_mark: 
- base fee + discount :heavy_check_mark: 
- custom fee :heavy_check_mark: 
- custom fee + discount :heavy_check_mark:

**Pull request merged**